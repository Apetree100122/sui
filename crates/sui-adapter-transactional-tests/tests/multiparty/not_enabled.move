// Copyright (c) Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0

//# init --addresses ex=0x0

//# publish
module ex::m;

public struct Pub has key, store {
    id: UID,
}

public struct Priv has key {
    id: UID,
}

public fun mint(ctx: &mut TxContext) {
    let p = Pub { id: object::new(ctx) };
    let q = Priv { id: object::new(ctx) };
    transfer::public_transfer(p, ctx.sender());
    transfer::transfer(q, ctx.sender());
}

public fun create_multiparty(ctx: &mut TxContext) {
    let p = Pub { id: object::new(ctx) };
    transfer::public_multiparty_transfer(p, vector[@0])
}

public fun pub_multiparty(obj: Pub, p: vector<address>) {
    transfer::public_multiparty_transfer(obj, p)
}

public fun priv_multiparty(obj: Priv, p: vector<address>) {
    transfer::multiparty_transfer(obj, p)
}

//# run ex::m::mint

// Aborts since multiparty transfer is not enabled
//# run ex::m::create_multiparty

// Aborts since multiparty transfer is not enabled
//# run ex::m::priv_multiparty --args object(2,0) vector[@0]

// Aborts since multiparty transfer is not enabled
//# run ex::m::pub_multiparty --args object(2,1) vector[@0]

// Aborts since multiparty transfer is not enabled
//# run sui::transfer::public_multiparty_transfer
//#     --type-args ex::m::Pub
//#     --args object(2,1) vector[@0]
