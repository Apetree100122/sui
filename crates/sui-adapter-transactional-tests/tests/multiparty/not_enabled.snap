---
source: external-crates/move/crates/move-transactional-test-runner/src/framework.rs
---
processed 7 tasks

task 1, lines 6-35:
//# publish
created: object(1,0)
mutated: object(0,0)
gas summary: computation_cost: 1000000, storage_cost: 7904000,  storage_rebate: 0, non_refundable_storage_fee: 0

task 2, lines 37-39:
//# run ex::m::mint
created: object(2,0), object(2,1)
mutated: object(0,0)
gas summary: computation_cost: 1000000, storage_cost: 3442800,  storage_rebate: 978120, non_refundable_storage_fee: 9880

task 3, lines 40-42:
//# run ex::m::create_multiparty
created: object(3,0)
mutated: object(0,0)
gas summary: computation_cost: 1000000, storage_cost: 2272400,  storage_rebate: 978120, non_refundable_storage_fee: 9880

task 4, lines 43-48:
//# programmable --inputs object(2,0) @0
//> 0: sui::multiparty::single_owner(Input(1));
//> ex::m::priv_multiparty(Input(0), Result(0))
// Aborts since multiparty transfer is not enabled
mutated: object(0,0), object(2,0)
gas summary: computation_cost: 1000000, storage_cost: 2280000,  storage_rebate: 2197008, non_refundable_storage_fee: 22192

task 5, lines 49-53:
//# programmable --inputs object(2,1) @0
//> 0: sui::multiparty::single_owner(Input(1));
//> ex::m::pub_multiparty(Input(0), Result(0))
// Aborts since multiparty transfer is not enabled
mutated: object(0,0), object(2,1)
gas summary: computation_cost: 1000000, storage_cost: 2272400,  storage_rebate: 2189484, non_refundable_storage_fee: 22116

task 6, lines 54-56:
//# programmable --inputs object(2,1) @0
//> 0: sui::multiparty::single_owner(Input(1));
//> sui::transfer::public_multiparty_transfer<ex::m::Pub>(Input(0), Result(0))
Error: Error checking transaction input objects: IncorrectUserSignature { error: "Object object(2,1) is owned by account address 0x0000000000000000000000000000000000000000000000000000000000000000, but given owner/signer address is 0xfccc9a421bbb13c1a66a1aa98f0ad75029ede94857779c6915b44f94068b921e" }
